### DISCUSSION TITLE
local using network activity

### DISCUSSION THREAD

**Discussion by d7zda** [ggml-org/llama.cpp#4290] [Q&A]
i am seeing network usage when i run the local llama.cpp on ubuntu, where can i find more info on this?
im looking for llm generators that dont use any network whatsoever

**Comment by bullno1** [comment]
What did you use to monitor? Does the network activity come from the process?

AFAIK, there's zero network code. server.cpp is an example and of course it uses network but that's local if you want to.

Edit: There's also MPI but isn't that kinda broken right now?

**Comment by d7zda** [comment]
i used the system monitor and just by eyeing the graph, after entering
prompts i see the bumps for the sending and receiving lines;
similar to #3999 , but im not sure how he resolved his issue.
https://github.com/ggerganov/llama.cpp/issues/3999
im a novice so idk much about technical analysis,

**Reply by cmp-nct** [reply]
It's certainly not from main or llava-cli
If you want to find out what is going on you need to look at the actual traffic in  tcpdump/windump/wireshark, there you'll see what is going on.
Best close the browser,mail etc during the test to clean it up as much as possible

**Reply by bullno1** [reply]
> after entering prompts i see the bumps for the sending and receiving lines;

That's not indicative of anything.

There are some docker images.
Put it in a no network namespace: `--network none`

Or even better, unplug your network and try.
